src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/HaFlowHistory.groovy

```diff
@@ -0,0 +1,48 @@
+package org.openkilda.functionaltests.helpers.model
+
+import org.openkilda.testing.service.northbound.NorthboundServiceV2
+
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.TupleConstructor
+import org.openkilda.testing.service.northbound.model.HaFlowHistoryEntry
+
+@TupleConstructor
+@EqualsAndHashCode
+class HaFlowHistory {
+    HaFlowHistory(List<HaFlowHistoryEntry> entries) {
+        this.entries = entries
+    }
+
+    HaFlowHistory(NorthboundServiceV2 northboundServiceV2, List<HaFlowHistoryEntry> entries) {
+        this.northboundServiceV2 = northboundServiceV2
+        this.entries = entries
+    }
+
+    NorthboundServiceV2 northboundServiceV2;
+    List<HaFlowHistoryEntry> entries;
+
+
+    HaFlowHistoryEntry[] getUpdateEntries(){
+        return entries.findAll({it ->it.action == "HA-Flow update"})
+    }
+
+    HaFlowHistoryEntry[] getCreateEntries(){
+        return entries.findAll({it ->it.action == "HA-Flow create"})
+    }
+
+    HaFlowHistoryEntry[] getDeleteEntries(){
+        return entries.findAll({it ->it.action == "HA-Flow delete"})
+    }
+
+    HaFlowHistoryEntry[] getRerouteEntries(){
+        return entries.findAll({it ->it.action == "HA-Flow reroute"})
+    }
```

Comment:
Please, add enum-based class HaFlowHistoryEntryType and merge  these four methods into one:
`HaFlowHistoryEntry[] getEntries(HaFlowHistoryEntryType  entryType)`
Also, are you sure that you want to work with Arrays, not Lists?

Reply:
done

Reply:
Not done. You may not operate on Strings when you have only several, pre-defined strings that can be used. Use [enum based class like here](https://github.com/telstra/open-kilda/blob/test/4925-multiple-opentsdb-targets/src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/model/stats/Direction.groovy). This would prevent API user from misprints and gives understanding what are the limitations on parameters, accepted by your method.

Reply:
done